### DISCUSSION TITLE
Row Component - 자식 컴포넌트 추가 및 left, right 지정, maxWidth

### DISCUSSION THREAD

**Discussion by healtheloper** [Co-Studo/Co-Studo-front#55] [Suggestions]
@mina-gwak 

### maxWidth

현재 Row Component 는 스터디의 정보를 보여주는 카드 컴포넌트에 종속적인 속성 maxWidth 를 가지고 있어
다양한 곳에 활용하기 위해선 해당 속성을 없애는게 좋다고 생각합니다.

Row Component 는 그 자체로 다양한 활용성이 있다고 생각이 드는데요, 
예를들어 div 태그에 display flex 를 넣어 사용하고 싶은 곳에서도 사용할 수 있다고 생각합니다.

### title, description

title, description 을 묶어서 사용할 수 있는 중간 컴포넌트(content) 를 지정하면 꼭 title, description 형태가 아니더라도 삼분할의 용도로 쓰일 수도 있다고 생각하구요

```tsx
<div css={{display: 'flex'}}>
  <span>안녕<span>
  <span>하세요</span>
</div>

<Row content={
  <>
    <span>안녕</span>
    <span>하세요</span>
  </>
} />
```

### title, description -> top, bottom

그럼 지금 쓰고있는 title, description 형태는 어떻게 할 것인지 생각해보면
Row 의 자식컴포넌트를 하나 선언하여 사용할 수도 있다고 생각합니다.

```tsx
Row.XXX = ({top, bottom}) => {
  <S.Main>
    {top}
    {bottom}
  </S.Main>
}

<Row content={<Row.XXX top={title} bottom={description} />} />
```

### Container flex layout

위와 같이 수정한다면 left, content, right 의 세 개의 컴포넌트가 남는데 
width 에서 각 요소의 길이들이 동일하면서 각 요소에 맞게 배치되어야 한다고 생각하기에 space-between 을 사용하는게 좋다고 생각합니다.

바로 수용 가능해서 작업 가능하신 부분이 있다면 말씀해주세요!
해당 글로는 의도 전달이 잘 안되는 것 같다면 Row2 등의 이름으로 제작해서 PR 날려보겠습니다

**Comment by jindonyy** [comment]
전~부 동의합니다! 근데 space-between을 하려면 저번에 말한거처럼 gap도 필요할텐데
이 부분 지난번에 미정했던거 같은데 맞나요..? 제가 prop 추가하지말고 `styled(Row)` 이런식으로 하자고 얘기했던거 같은뎅..!

**Reply by healtheloper** [reply]
각 컴포넌트 간의 간격(gap) 이 Row 를 사용할 때 기대할 수 있는 기본 동작이라고 생각했어요
기능만을 제공하는 컴포넌트로 분리될 필요가 없는 컴포넌트라 styled 함수를 통해 스타일을 입혀주는 과정은 딱히 필요하지 않다고 생각합니다

**Comment by mina-gwak** [comment]
넵 대부분 동의합니다! 마지막에 작성하신 부분에서 처음 row 컴포넌트로 분리해야겠다고 생각하게 했던 아래와 같은 레이아웃은 space-between을 사용했을 때 어떤 식으로 구현하실 예정인지 궁금하네요. 일단 파크가 생각하신대로 진행해보고 이 부분은 그때 다시 이야기해봐도 좋을 것 같습니다!

<img width="216" alt="193102656-d0a6af5f-8237-4b55-aeac-f91bc923a17e" src="https://user-images.githubusercontent.com/62706988/196284063-04ad71d8-c737-46ff-aff7-f80d4746c18f.png">
